sirepo/pkcli/admin.py

```diff
@@ -93,6 +95,29 @@ def create_examples():
                     _create_example(example)
 
 
+def setup_dev_proprietary_code(sim_type, rpm_url):
+    """Get an rpm and put it in the proprietary code dir for a sim type.
+
+    Args:
+      sim_type (str): simulation type
+      rpm_url (str): Url of the rpm (file:// or http://)
+    """
+    import sirepo.pkcli.admin
+    import urllib.request
+
+    assert pkconfig.channel_in('dev'), \
+        'Only to be used in dev. channel={}'.format(pkconfig.cfg.channel)
+
+    d = sirepo.pkcli.admin.proprietary_code_dir(sim_type)
+    pkio.mkdir_parent(d)
+    s = sirepo.sim_data.get_class(sim_type)
+
+    urllib.request.urlretrieve(
+        rpm_url,
+        d.join(getattr(s, '{}_RPM'.format(sim_type.upper()))),
```

Comment:
clever, but maybe add an entrypoint in sim_data called `proprietary_code_rpm` that returns the name so that you don't have to do this manipulation. This would blow up for any non-proprietary code.

Reply:
https://github.com/radiasoft/sirepo/pull/2490/commits/cd07af6a29565ca8f4b5a3144ff997244442b7ea